src/instamatic/camera/camera_serval.py

```diff
@@ -46,12 +67,32 @@ def get_image(self, exposure=None, binsize=None, **kwargs) -> np.ndarray:
         """
         if exposure is None:
             exposure = self.default_exposure
-        if not binsize:
-            binsize = self.default_binsize
-
+        if exposure < self.MIN_EXPOSURE:
+            logger.warning(f'{self.BAD_EXPOSURE_MSG}: {exposure}')
```

Comment:
Consider deferring the formatting until necessary: https://docs.python.org/3/howto/logging.html#optimization

```suggestion
            logger.warning('%s: %d', self.BAD_EXPOSURE_MSG, exposure)
```